app/models/manageiq/providers/ibm_cloud/inventory/collector/power_virtual_servers.rb

```diff
@@ -3,47 +3,92 @@ class ManageIQ::Providers::IbmCloud::Inventory::Collector::PowerVirtualServers <
   require_nested :NetworkManager
   require_nested :StorageManager
 
-  def connection
-    @power_iaas ||= manager.connect(:service => "PowerIaas")
+  def pvm_instances
+    @pvm_instances ||= pvm_instances_api.pcloud_pvminstances_getall(cloud_instance_id).pvm_instances || []
   end
 
-  def vms
-    connection.get_pvm_instances
+  def pvm_instance(instance_id)
+    pvm_instances_api.pcloud_pvminstances_get(cloud_instance_id, instance_id)
   end
 
   def image(img_id)
-    connection.get_image(img_id)
+    images_api.pcloud_cloudinstances_images_get(cloud_instance_id, img_id)
   end
 
   def images
-    connection.get_images
+    @images ||= images_api.pcloud_cloudinstances_images_getall(cloud_instance_id).images || []
   end
 
   def volumes
-    connection.get_volumes
+    @volumes ||= volumes_api.pcloud_cloudinstances_volumes_getall(cloud_instance_id).volumes || []
   end
 
   def volume(volume_id)
-    connection.get_volume(volume_id)
+    volumes_api.pcloud_cloudinstances_volumes_get(cloud_instance_id, volume_id)
   end
 
   def networks
-    connection.get_networks
+    @networks ||= networks_api.pcloud_networks_getall(cloud_instance_id).networks || []
+  end
+
+  def network(network_id)
+    networks_api.pcloud_networks_get(cloud_instance_id, network_id)
   end
 
   def ports(network_id)
-    connection.get_network_ports(network_id)
+    @ports ||= networks_api.pcloud_networks_ports_getall(cloud_instance_id, network_id).ports || []
   end
 
   def sshkeys
-    connection.get_ssh_keys
+    @ssh_keys ||= tenants_api.pcloud_tenants_get(tenant_id).ssh_keys
   end
 
-  def system_pool
-    connection.get_system_pool.values
+  def system_pools
+    @system_pools ||= system_pools_api.pcloud_systempools_get(cloud_instance_id)
   end
 
   def storage_types
-    connection.get_storage_types
+    # TODO
```

Comment:
I don't think the [/broker/v1/storage-types](https://github.com/IBM-Cloud/ibm-cloud-sdk-ruby/blob/master/lib/ibm/cloud/sdk/power_iaas.rb#L169) is documented anywhere and I'm not sure when it will be added to the Power Cloud API.

is it a bad idea to a) manually patch the generated SDK b) make a `RestClient` call directly from here?

Reply:
Yeah I was looking for this on the API docs site and didn't see anything, would like to know if this is something that will be added to the openapi spec in the future or were living dangerously using undocumented apis 😆

Reply:
A new API call should be added in the future to return storage availability (I'm guessing that it will be similar to [PCloudSystemPools](https://cloud.ibm.com/apidocs/power-cloud#pcloud-systempools-get)). The return data structure might change, this [/broker/v1/storage-types](https://github.com/IBM-Cloud/ibm-cloud-sdk-ruby/blob/master/lib/ibm/cloud/sdk/power_iaas.rb#L169) call could disappear, I'm not sure - so yeah this could be kind of risky. I can get more details next week.

If you think it's a bad idea to use this interim call we could just use a hard coded list of storage types. Not all types are available in all regions so it could cause some volume provisioning fails.

Reply:
> this /broker/v1/storage-types call could disappear

Yeah this is the part that worries me, if we release with this and it goes away refreshes will fail (or at least we won't be pulling storage-types and provisions wouldn't work) until the customer upgraded.  If the intention is to deprecate this API but isn't completed by the time we release I think a hard coded list would be safest.

We don't need to solve this in this PR, I'm going to replace with a single RestClient call for now.